### PR TITLE
Add dateFormat trait

### PR DESCRIPTION
**Pull request by daddykotex**
Let's start with that, no format for now.

### CODE REVIEW

**Comment by Baccata** [comment]
Maybe a rename would be wise ? `dateFormat` rather than `dateOnly` (which was initially justified due to targeting a timestamp rather than a string)

**Comment by daddykotex** [comment]
> Maybe a rename would be wise ? `dateFormat` rather than `dateOnly` (which was initially justified due to targeting a timestamp rather than a string)

Ok, that makes sense to me.

**Review by Baccata — APPROVED** [review]

**Review by lewisjkl — APPROVED** [review]

File: modules/core/resources/META-INF/smithy/formats.smithy [hunk 1]
```diff
@@ -0,0 +1,10 @@
+$version: "2"
+
+namespace alloy
+
+/// This trait indicates that a String value contains a date without
+/// a time component.
+/// e.g.: "2022-03-30"
```
Comment (kubukoz) [inline modules/core/resources/META-INF/smithy/formats.smithy:7]: should this refer to some specific name from ISO8601?

Comment (Baccata) [reply modules/core/resources/META-INF/smithy/formats.smithy:7]: sure

Comment (daddykotex) [reply modules/core/resources/META-INF/smithy/formats.smithy:7]: will fix

Comment (daddykotex) [reply modules/core/resources/META-INF/smithy/formats.smithy:15]: Updated with info from RFC 3339: https://www.rfc-editor.org/rfc/rfc3339#section-5.6

**Review by Baccata — APPROVED** [review]

**Pull request merged**